observability/WFLY-20296-add-support-for-more-opentelemetry-config-options.adoc

```diff
@@ -0,0 +1,154 @@
+** The shape of the management model should be flat, using a list of attributes, rather than nested resources
+*** Each attribute will be declared on the existing `OpenTelemetrySubsystemRegistrar`. No new child resources should be created.
+*** Attribute groups will be used to group properties by signal type to make the XML file and admin console more user-friendly
+** Shared properties (e.g., the OpenTelemetry Collector endpoint) should be exposed on the root element.
+*** Signal-specific overrides need not be supported
+* All new properties added will have, as defaults, values described in the relevant OpenTelemetry documentation (see the official docs https://opentelemetry.io/docs/languages/java/configuration/[here])
+* Support should not depend on, e.g., MicroProfile Telemetry or MicroProfile Config
+* Any configured properties should still be able to be overridden via the existing MicroProfile support (this is the current default behavior)
+
+[IMPORTANT]
+=====
+Open question: Some of the values specified by OpenTelemetry are rather large and unwieldy (e.g., `base2_exponential_bucket_histogram`). Should we provide shorthand mappings, use the exact strings from otel docs, or both?
+=====
+
+=== Changed requirements
+
+N/A
+
+=== Non-Requirements
+
+* The OpenTelemetry documentation lists several signal-specific parameters to configure each signal explicitly. The following parameters have the same defaults as those already supported (i.e., for traces), so, in the name of simplicity, these will be configured implicitly by the system using a single configuration attribute:
+** `otel.blrp.max.queue.size` via `max-queue-size`
+** `otel.blrp.max.export.batch.size` via `max-export-batch-size`
+** `otel.blrp.export.timeout` via `export-timeout`
+* The SmallRye OpenTelemetry OTLP exporters do not support `otel.java.exporter.memory_mode`, so it will not be exposed.
+* While unifying observability configuration (i.e., that of the OpenTelemetry and Micrometer subsystems) has been requested and might be of interest to users and administrators, that is a non-goal for this RFE.
+
+=== Future Work
+
+* Should users request fine-grained, per-signal configuration, we can revisit the shared attributes mentioned above.
+
+== Backwards Compatibility
+
+This change will, of necessity, change the management model of the subsystem. Despite this change, existing configurations should continue to work unchanged. Furthermore, existing configurations should be automatically migrated to the new model/schema once a relevant change has made to the configuration.
+
+=== Default Configuration
+
+No default values will be changed.
+
+=== Importing Existing Configuration
+
+Existing configurations should be supported unchanged.
+
+=== Deployments
+
+Deployments should see no change in behavior.
+
+=== Interoperability
+
+No interoperability concerns are foreseen.
+
+== Implementation Plan
+
+N/A
+
+== Admin Clients
+
+N/A
```

Comment:
For sure it's applicable. Needs a statement, even if it's just that the CLI and HAL will just work with no changes on their side.